x/incentives/keeper/store_test.go

```diff
@@ -33,3 +38,45 @@ func (s *KeeperTestSuite) TestGaugeReferencesManagement() {
 	gaugeRefs3 := s.App.IncentivesKeeper.GetGaugeRefs(s.Ctx, key2)
 	s.Require().Equal(len(gaugeRefs3), 2)
 }
+
+func (s *KeeperTestSuite) TestGetGroupGaugeById() {
+	tests := map[string]struct {
+		groupGaugeId   uint64
+		expectedRecord types.GroupGauge
+	}{
+		"Valid record": {
+			groupGaugeId: uint64(5),
+			expectedRecord: types.GroupGauge{
+				GroupGaugeId: uint64(5),
+				InternalIds:  []uint64{2, 3, 4},
+			},
+		},
+
+		"InValid record": {
+			groupGaugeId:   uint64(6),
+			expectedRecord: types.GroupGauge{},
+		},
+	}
+
+	for name, test := range tests {
+		s.Run(name, func() {
+			s.FundAcc(s.TestAccs[1], sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100_000_000)))) // 1,000 osmo
+			clPool := s.PrepareConcentratedPool()                                                 // gaugeid = 1
+
+			// create 3 internal Gauge
+			var internalGauges []uint64
+			for i := 0; i <= 2; i++ {
+				internalGauge := s.CreateNoLockExternalGauges(clPool.GetId(), sdk.NewCoins(), s.TestAccs[1], uint64(1)) // gauge id = 2,3,4
```

Comment:
For the sake of more readable & robust tests, it would probably be better to abstract this setup into a test helper and add `numberOfExistingGauges` as a test parameter

Reply:
created helper for creating internalGauges for GroupGauges
https://github.com/osmosis-labs/osmosis/pull/6085/commits/c99cb1de19308284b9f84184660245572de9bdda#diff-1000dbecb1333fe09226de57d089fe167e289bc9edae3aac2f040029bfa91b1eR51